### ISSUE TITLE
A semantic mistake in SVG 

### ISSUE THREAD

**Issue by dborisog** [ocsigen/tyxml#250]

**Comment by Drup** [comment]
Hmm, the specifications for SVG 2 and SVG 1.1 seem inconsistent. SVG2 refers to CSS transforms, which mandates use of the unit, while SVG 1.1 uses its own format, which doesn't have units. Seems like chrome doesn't use units.

How does it work with other browsers ?

**Comment by dborisog** [comment]
Yes, the behavior is the same, both generated and removal of "deg" from "270deg" in DevTools, in
* Chrome, 79.0.3945.88
* Firefox, 71.0
* Opera, 64.0.3417.92
* (I do not have Yandex browser and Microsoft Edge on my Ubuntu 18.04 64 bit PC).

**Comment by Drup** [comment]
ok, so I suggest we get rid of all the units for rotate, keep only a float, and link to the msdn documentation. Are you willing to propose a patch ?

**Comment by dborisog** [comment]
I think I need at least a half of a year's experience with Ocsigen's architecture and functional programming in OCaml in general prior to any pull requests to major projects.

The proposed solution might work, but I have limited experience with one case only. I used a modified Rotate example from ( https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform ) in ( https://www.w3schools.com/graphics/tryit.asp?filename=trysvg_text2 ) prior to opening this issue. I do not know how it would affect other Transform functions.

Most likely, SVG use is very limited at the moment, thus a narrow subset of Ocsigen users if any will be affected by the change.

**Comment by Drup** [comment]
After more research : When `None` is provided for units, nothing is printed, which is exactly what is expected in SVG 1.1. SVG 2, on the contrary, uses the same syntax as CSS, which mandates units.

In conclusion, I propose to not change the API at all: you should simply give `None` as units to ensure compatibility with current browsers.

**Comment by dborisog** [comment]
My ability to judge is currently paralyzed by incomplete understanding of category theory. Category theory described mathematics from the perspective of changes of mathematical objects, which allows focusing on very important properties of mathematical structures.

In category theory, we have objects transformed into other objects by morphisms. This is very fitting for functional programming, where we transform one data type into another data type. The end-result of TyXML transformations is an HTML/SVG/XML document.

Category theory has multiple categories described, and it also has _universal constructions_, including _initial object_ and _terminal object_. Bartosz Milewski (see pp. 52--55 https://github.com/hmemcpy/milewski-ctfp-pdf ), while providing examples on Haskell, states that _Void_ stands for _initial object_ and _unit_ stands for _terminal object in Haskell. It might be very true for _None_ and _unit_ in OCaml. It might be important, as TyXML transformations fitting to the math of category theory could allow smooth construction of powerful (semi)polymorphic systems of functions.

For my incomplete understanding of category theory, I simply cannot comment on the rights/wrongs of the behavior you have described earlier. I am currently fixing my intellectual deficiency, with an intention to write a category-theory-driven data transformations in Ocsigen. I intend to comment on this issue as soon as I have sufficient theoretical knowledge and practical expertise. 

Meanwhile, unless some words in this comment provided useful insights, the absence of changes on a thing that is already working (with PPX and functional workarounds) seems the right thing to do.
